src/main/argv.ts

```diff
@@ -152,4 +153,38 @@ export const getStartupApp = (arg: typeof argv): StartupApp | undefined => {
     }
 };
 
+const removeArguments = (args: string[], argumentsToRemove: string[]) => {
+    const result = [...args];
+
+    argumentsToRemove.forEach(argToRemove => {
+        const index = result.findIndex(arg => arg === argToRemove);
+        if (index !== -1) {
+            result.splice(index, 2);
+        }
+    });
+
+    return result;
```

Comment:
No, because e.g. in the test case https://github.com/NordicSemiconductor/pc-nrfconnect-launcher/blob/a96f478ebad6ec4e0299cfbbea4f10f5d0e6831a/src/main/argv.test.ts#L16 that would only remove the `--deviceSerial` not also the argument `001122`. But I was also unsure whether this behaviour is obvious enough in the code with the 2 in `splice(index, 2)`. At least it is demonstrated in the test.

Reply:
I now put the `2` in a constant `argumentCountToRemove`, to make this more explicit.